src/components/molecules/Button/Button.tsx

```diff
@@ -0,0 +1,77 @@
+import { FC, PropsWithChildren, SVGProps, FunctionComponent } from 'react'
+import BaseButton, {
+  BaseButtonProps,
+} from 'components/atoms/BaseButton/BaseButton'
+import classNames from 'classnames'
+import classes from './styles.module.scss'
+
+export enum AppearanceTypes {
+  Primary = 'primary',
+  Secondary = 'secondary',
+  Text = 'text',
+}
+
+export enum SizeTypes {
+  M = 'm',
+  S = 's',
+}
+
+export interface ButtonProps extends BaseButtonProps {
+  appearance?: AppearanceTypes
+  size?: SizeTypes
+  icon?: FunctionComponent<SVGProps<SVGSVGElement>>
+  ariaLabel?: string
+  hidden?: boolean
+  disabled?: boolean
+  className?: string
+}
+
+export type IconProps = {
+  icon?: FunctionComponent<SVGProps<SVGSVGElement>>
+  className?: string
+  ariaLabel?: string
+}
+
+const Icon: FC<IconProps> = ({ icon: IconComponent, className, ariaLabel }) => {
+  const ariaLabelToUse = ariaLabel
+
+  if (!IconComponent) return null
+  return (
+    <IconComponent
+      className={(classes.icon, className)}
```

Comment:
**[blocker]**

1. classes.icon doesn't seem to exist
2. construction a class name like this won't work. Classname should be a string, not (string, string). Even if classes.icon would exist, it wouldn't be applied like this